contracts/tokens/MetaStudioToken.sol

```diff
@@ -1,42 +1,55 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.9;
 
-import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/draft-ERC20PermitUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20VotesUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/interfaces/IERC165Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/interfaces/IERC777Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/interfaces/IERC1820ImplementerUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC777/ERC777Upgradeable.sol";
 import "../metatx/ERC2771ContextUpgradeable.sol";
 import "../metatx/IERC2771Upgradeable.sol";
 
+/// @title The Metastudio's ERC777/ERC20 token
 /// @custom:security-contact it@theblockchainxdev.com:
 contract MetaStudioToken is
   IERC165Upgradeable,
   Initializable,
-  ERC20Upgradeable,
+  ReentrancyGuardUpgradeable,
   OwnableUpgradeable,
   PausableUpgradeable,
   ERC20PermitUpgradeable,
   ERC20VotesUpgradeable,
-  UUPSUpgradeable,
-  ERC2771ContextUpgradeable
+  ERC777Upgradeable,
+  IERC1820ImplementerUpgradeable,
+  ERC2771ContextUpgradeable,
+  UUPSUpgradeable
 {
   /// @custom:oz-upgrades-unsafe-allow constructor
   constructor() {
     _disableInitializers();
   }
 
-  function initialize(address tokensOwner, address forwarder)
-    external
-    initializer
-  {
+  /// @notice Contract initialisation. 5_000_000_000 tokens are minted
+  /// @dev Constructor replacement methods used for Proxified Contract
+  /// @param tokensOwner initianally minted Token's owner address
+  /// @param forwarder Initial ERC2771 trusted forwarder
+  /// @param defaultOperators_ Array of default operators for ERC777
+  function initialize(
+    address tokensOwner,
+    address forwarder,
+    address[] memory defaultOperators_
```

Comment:
Can we use the same pattern for the variables ?
 tokensOwner <> tokensOwner_ ?
defaultOperators_ <> defaultOperators

Reply:
Nassim read that suffix "_" means array

Reply:
I can't find that information
I know that some people prefix tthe variables to not shadow other global varibles
```
constructor(address _token) {
  token = _token
}
```